datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -183,28 +290,32 @@ protected Object getFailurePayload(SdxDetachSuccessEvent payload, Optional<SdxCo
     public Action<?, ?> failedAction() {
         return new AbstractSdxAction<>(SdxDetachFailedEvent.class) {
             @Override
-            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+            protected SdxContext createFlowContext(FlowParameters flowParameters,
+                    StateContext<FlowState, FlowEvent> stateContext,
                     SdxDetachFailedEvent payload) {
                 return SdxContext.from(flowParameters, payload);
             }
 
             @Override
-            protected void doExecute(SdxContext context, SdxDetachFailedEvent payload, Map<Object, Object> variables) throws Exception {
+            protected void doExecute(SdxContext context, SdxDetachFailedEvent payload,
+                    Map<Object, Object> variables) throws Exception {
                 Exception exception = payload.getException();
-                DatalakeStatusEnum failedStatus = DatalakeStatusEnum.STOPPED;
-                LOGGER.error("Detaching SDX: {} failed with status {}", payload.getSdxName(), failedStatus, exception);
+                LOGGER.error("Detaching SDX cluster with ID {} and name {} failed with error {}.",
+                        payload.getResourceId(), payload.getSdxName(), exception.getMessage(), exception);
+
                 String statusReason = "SDX detach failed";
                 if (exception.getMessage() != null) {
                     statusReason = exception.getMessage();
                 }
-                sdxStatusService.setStatusForDatalakeAndNotify(failedStatus, statusReason, payload.getResourceId());
-                Flow flow = getFlow(context.getFlowParameters().getFlowId());
-                flow.setFlowFailed(payload.getException());
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.STOPPED, statusReason, payload.getResourceId());
+
+                getFlow(context.getFlowParameters().getFlowId()).setFlowFailed(payload.getException());
                 sendEvent(context, SDX_DETACH_FAILED_HANDLED_EVENT.event(), payload);
             }
 
             @Override
-            protected Object getFailurePayload(SdxDetachFailedEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+            protected Object getFailurePayload(SdxDetachFailedEvent payload, Optional<SdxContext> flowContext,
+                    Exception e) {
                 return null;
```

Comment:
could you add a log here? I have seen really rarely that failure handling actions throw an error

Reply:
Done.